Library/Homebrew/linkage_checker.rb

```diff
@@ -196,7 +196,14 @@ def check_dylibs(rebuild_cache:)
         rescue Errno::ENOENT
           next if harmless_broken_link?(dylib)
 
-          if (dep = dylib_to_dep(dylib))
+          basename = File.basename(dylib)
+          if dylib_found_via_dlopen(basename)
```

Comment:
Doing this first does change the output of `brew linkage` slightly. System dylibs (e.g. `libSystem.B.dylib`) will typically be reported with only their basename, but system frameworks (e.g. `CoreServices`) need to be found via an absolute path so one will be reported.

This is probably more meaningful anyway, since these paths are no longer on-disk.